### PR TITLE
Add uniqueness on question name depending on user_id

### LINKED ISSUE #196: Question name unique by user
`User 1` and `User 2` can have one `question` with the same `name` ,
`User` can't have 2 questions with same name

### PR DESCRIPTION
**Pull request by garciajordy**
# QuestionScope

**QuestionModel:**
- Add uniqueness to name with user scope.

**Schema:**
- Add index on name with user_id scope.

**Rspec:**
- Add tests to check if uniqueness works depending on which user creates the question with the same name.

_Thanks for taking the time to review my PR!_ 😄 
_It was fun to learn more about this!_
fix #196

### CODE REVIEW

File: backend/.ruby-version [hunk 1]
```diff
@@ -1 +1 @@
-3.0.1
+3.0.0
```
Comment (edimossilva) [inline backend/.ruby-version:1]: you should not change this

File: backend/Gemfile [hunk 2]
```diff
@@ -1,7 +1,7 @@
 source 'https://rubygems.org'
 git_source(:github) { |repo| "https://github.com/#{repo}.git" }
 
-ruby '3.0.1'
+ruby '3.0.0'
```
Comment (edimossilva) [inline backend/Gemfile:4]: you should not change this

File: backend/Gemfile.lock [hunk 3]
```diff
@@ -433,7 +433,7 @@ DEPENDENCIES
   webpacker (~> 5.0)
 
 RUBY VERSION
-   ruby 3.0.1p64
+   ruby 3.0.0p0
```
Comment (edimossilva) [inline backend/Gemfile.lock:436]: you should not change this

File: backend/spec/models/question_spec.rb [hunk 4]
```diff
@@ -33,4 +35,9 @@
     it { expect(question_single.single_choice?).to eq(true) }
     it { expect(question_multiple.single_choice?).to eq(false) }
   end
+
+  describe 'uniqueness by user' do
```
Comment (edimossilva) [inline backend/spec/models/question_spec.rb:39]: you can use some ready code from `shoulda matcher`gem

**Review by edimossilva — CHANGES_REQUESTED** [review]
Also link the issue to the PR

**Comment by garciajordy** [comment]
# Changes

**Rspec:**
- Use the shoulda-matchers helpers.

**GemFile:**
- Change back to 3.0.1.

**Review by edimossilva — APPROVED** [review]
Approved, can you link the issue to this PR and merge it?

**Pull request merged**